relay-server/src/services/health_check.rs

```diff
@@ -149,11 +149,18 @@ impl HealthCheckService {
     }
 
     async fn spool_health_probe(&self) -> Status {
-        let has_capacity = self
-            .envelope_buffer
-            .as_ref()
-            .map_or(true, |b| b.has_capacity());
-        match has_capacity {
+        let buffers = self.partitioned_buffer.buffers();
+        // If no buffer is supplied, we assume it's healthy.
+        let all_have_capacity = if buffers.is_empty() {
+            true
+        } else {
+            self.partitioned_buffer
+                .buffers()
```

Comment:
Is this method used anywhere else? If not, I would give `PartitionedEnvelopeBuffer` a `has_capacity` method instead.